tests/e2e/csi_static_provisioning_basic.go

```diff
@@ -1686,4 +1716,154 @@ var _ = ginkgo.Describe("Basic Static Provisioning", func() {
+				pvc = nil
+
+				ginkgo.By("PV will be in released state , hence delete PV explicitly")
+				framework.ExpectNoError(fpv.DeletePersistentVolume(client, pv.GetName()))
+				pv = nil
+
+				ginkgo.By("Verify CRD should be deleted automatically")
+				//TODO add polling wait
+				time.Sleep(time.Duration(40) * time.Second)
+				flag := queryCNSRegisterVolume(ctx, restConfig, cnsRegisterVolumeName, namespace)
+				gomega.Expect(flag).NotTo(gomega.BeTrue())
+
+				ginkgo.By("Delete Resource quota")
+				deleteResourceQuota(client, namespace)
+			}()
+		}
+
+	})
+
+	/*
+		Specify VolumeID and DiskURL together and verify the error message
+		Presetup: Create VMDK with valid storage policy.
+		Test Steps:
+		1. Create FCD
+		2. Create Resource quota
+		3. Create CNS register volume with above created VMDK and FCDID
+		4. Verify the error message "VolumeID and DiskURLPath cannot be specified together"
+	*/
+	ginkgo.It("[csi-supervisor] Specify VolumeID and DiskURL together and verify the error message", func() {
+
+		var err error
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		dataStoreType, err := defaultDatastore.Type(ctx)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		framework.Logf("Datastore type: %s", dataStoreType)
+		if dataStoreType != "vsan" {
+			ginkgo.Skip("Skipping 'Specify VolumeID and DiskURL together and verify the error message' test since the testbed does not have vSAN datastore")
+		} else {
+			curtime := time.Now().Unix()
+			randomValue := rand.Int()
+			val := strconv.FormatInt(int64(randomValue), 10)
+			val = string(val[1:3])
+			curtimestring := strconv.FormatInt(curtime, 10)
+			pvcName := "cns-pvc-" + curtimestring + val
+			framework.Logf("pvc name :%s", pvcName)
+			namespace = getNamespaceToRunTests(f)
+
+			restConfig, storageclass, profileID := staticProvisioningPreSetUpUtilForVMDKTests(ctx)
+
+			defer func() {
+				framework.Logf("Delete storage class")
+				err = client.StorageV1().StorageClasses().Delete(ctx, storageclass.Name, metav1.DeleteOptions{})
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}()
+
+			ginkgo.By("Creating FCD Disk")
+			fcdID, err := e2eVSphere.createFCDwithValidProfileID(ctx, "staticfcd"+curtimestring+val, profileID, diskSizeInMb, defaultDatastore.Reference())
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			framework.Logf("fcdID  : %s", fcdID)
+			deleteFCDRequired = false
+
+			vmdk := GetAndExpectStringEnvVar(envVmdkDiskURL)
+			framework.Logf("VMDK path : %s", vmdk)
+			ginkgo.By("Create CNS register volume with VMDK")
+			cnsRegisterVolume := getCNSRegisterVolumeSpec(ctx, namespace, fcdID, vmdk, pvcName, v1.ReadWriteOnce)
+			err = createCNSRegisterVolume(ctx, restConfig, cnsRegisterVolume)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			framework.Logf("waiting for some time for FCD to register in CNS and for cnsRegisterVolume to get create")
+			//TODO add polling wait
```

Comment:
Can you create an issue on github for this //TODO:
There are 2 places where you need polling according to this PR. We need to address this in future. Creating an issue for tracking will be useful

Reply:
Added poling instead of static wait .